src/index.js

```diff
@@ -220,8 +223,20 @@ class Masq {
 
   requestMasqAccessDone () {
     return new Promise((resolve, reject) => {
-      if (this._requestMasqAccessDone) return resolve()
-      this._onRequestMasqAccessDone = resolve
+      if (this._requestMasqAccessDone) {
+        if (this._requestMasqAccessErr) {
+          reject(this._requestMasqAccessErr)
+        } else {
+          resolve()
+        }
+      }
+      this._onRequestMasqAccessDone = () => {
```

Comment:
why would you need to define requestMasqAccessDone and _requestMasqAccessDone ? they share the exact same code

Reply:
requestMasqAccessDone will be resolved immediately if the request Access is done (_requestAccessDone has been set to true).
In the case it wasn't done, we have to wait until it is done to resolve the promise, that's why we set a function that calls the resolve/reject functions.